### PR TITLE
Update server path and payload

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Adapt code to SKWebAPI (as opposed to KernelHttpServer)

### Description
Change Web App to reflect new backend
Put in way to set port on backend

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/packaged/CopilotChat/SKWebApi/Program.cs [hunk 1]
```diff
@@ -62,7 +71,7 @@ private static void AddSemanticKernelServices(this IServiceCollection services,
         kernelConfig.AddCompletionBackend(completionConfig);
         ISemanticTextMemory memory = NullMemory.Instance;
         AIServiceConfig embeddingConfig = configuration.GetSection("EmbeddingConfig").Get<AIServiceConfig>();
-        if (embeddingConfig.IsValid())
+        if (embeddingConfig?.IsValid() == true)
```
Comment (craigomatic) [inline samples/packaged/CopilotChat/SKWebApi/Program.cs:74]: Does IsValid() not already allow the true/false comparison?

File: samples/packaged/CopilotChat/WebApp/README.md [hunk 2]
```diff
@@ -2,11 +2,7 @@
 
 ## Running the sample
 
-1. You will need an
-   [Azure Open AI Service key and endpoint](https://learn.microsoft.com/azure/cognitive-services/openai/quickstart)
-   for this sample
-2. Ensure the KernelHttpServer sample is already running at `http://localhost:7071`. If not, follow the steps 
-   to start it [here](../../dotnet/KernelHttpServer/README.md).
+1. Ensure the SKWebApi running at `https://localhost:40443`.d).
```
Comment (craigomatic) [inline samples/packaged/CopilotChat/WebApp/README.md:5]: Looks like something funny is happening at the end of this line?

File: samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        fetchAsync();
+    });
+
+    return (
+        <div style={{ padding: 80, gap: 20, display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
+            <Title3>Looking for your function</Title3>
+            <Spinner />
+            <Body1>
+                This sample expects to find the Azure Function from <strong>samples/dotnet/KernelHttpServer</strong>{' '}
```
Comment (craigomatic) [inline samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx:32]: Pointing to the wrong sample host?

File: samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                <a
+                    href="https://learn.microsoft.com/azure/azure-functions/functions-develop-vs-code?tabs=csharp"
+                    target="_blank"
+                    rel="noreferrer"
+                >
+                    Visual Studio Code
+                </a>{' '}
+                or from the command line using the{' '}
+                <a href="https://github.com/Azure/azure-functions-core-tools" target="_blank" rel="noreferrer">
+                    Azure Functions Core Tools
```
Comment (craigomatic) [inline samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx:54]: Don't need the function tools, just need dotnet 6, right?

**Review by craigomatic — CHANGES_REQUESTED** [review]
Broadly looks good, a few minor comments to resolve

File: samples/packaged/CopilotChat/SKWebApi/SKWebApiConstants.cs [hunk 5]
```diff
@@ -2,7 +2,8 @@
 
 namespace SKWebApi;
```
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/SKWebApi/SKWebApiConstants.cs:3]: Let's update this namespace to be consistent with the rest of the files.

File: samples/packaged/CopilotChat/SKWebApi/Program.cs [hunk 6]
```diff
@@ -6,6 +6,7 @@
 using Microsoft.SemanticKernel.SkillDefinition;
 using Microsoft.SemanticKernel.TemplateEngine;
 using SemanticKernel.Service.Config;
+using SKWebApi;
```
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/SKWebApi/Program.cs:9]: We should update whatever still has this namespace to what is being used here (i.e. SemanticKernel.Service)

File: samples/packaged/CopilotChat/SKWebApi/Program.cs [hunk 7]
```diff
@@ -17,6 +18,14 @@ public static void Main(string[] args)
 
         builder.Host.ConfigureAppSettings();
 
+        // Set port to run on
+        string serverPortString = builder.Configuration.GetSection("ServicePort").Get<string>();
+        if (!int.TryParse(serverPortString, out int serverPort))
```
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/SKWebApi/Program.cs:23]: Should we set a default serverPort or fail if one is not found?

File: samples/packaged/CopilotChat/SKWebApi/appsettings.json [hunk 8]
```diff
@@ -1,4 +1,5 @@
 {
+  "ServicePort": "", // Ex: "5000" - Defaults to "40443"
```
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/SKWebApi/appsettings.json:2]: If there is a true default that occurs, let's put it as the default value here.

File: samples/packaged/CopilotChat/WebApp/README.md [hunk 9]
```diff
@@ -16,11 +12,7 @@
 4. Create an **[.env](.env)** file to this folder root with the following variables and fill in with your information, where `REACT_APP_GRAPH_CLIENT_ID=` is the GUID copied from the **Application (client) ID** from the Azure Portal:
         
         VITE_REACT_APP_CHAT_CLIENT_ID=
-        VITE_REACT_APP_FUNCTION_URI=http://localhost:7071
-        VITE_REACT_APP_AZURE_OPEN_AI_KEY=
-        VITE_REACT_APP_AZURE_OPEN_AI_COMPLETION_DEPLOYMENT=
-        VITE_REACT_APP_AZURE_OPEN_AI_EMBEDDINGS_DEPLOYMENT=
-        VITE_REACT_APP_AZURE_OPEN_AI_ENDPOINT=
+        VITE_REACT_APP_FUNCTION_URI=http://localhost:40443
```
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/WebApp/README.md:15]: Should we add a comment to make sure this matches with what's in the SKWebApi appsettings?

File: samples/packaged/CopilotChat/WebApp/README.md (hunk 2 above)
Comment (adrianwyatt) [inline samples/packaged/CopilotChat/WebApp/README.md:5]: We should have a similar README for getting the SKWebApi up and running if we don't already (and reference it here).

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/packaged/CopilotChat/SKWebApi/Program.cs (hunk 1 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/SKWebApi/Program.cs:74]: It does, but notice the "?".  This checks whether we are null, true, or false

File: samples/packaged/CopilotChat/WebApp/README.md (hunk 2 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/WebApp/README.md:5]: Thanks for catching this.  Fixed

File: samples/packaged/CopilotChat/SKWebApi/appsettings.json (hunk 8 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/SKWebApi/appsettings.json:2]: Done

File: samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx (hunk 3 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx:32]: Fixed.  Thanks!

File: samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx (hunk 4 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/WebApp/src/components/views/FunctionProbe.tsx:54]: Fixed!

File: samples/packaged/CopilotChat/WebApp/README.md (hunk 9 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/WebApp/README.md:15]: Addressed

File: samples/packaged/CopilotChat/SKWebApi/SKWebApiConstants.cs (hunk 5 above)
Comment (glahaye) [reply samples/packaged/CopilotChat/SKWebApi/SKWebApiConstants.cs:3]: Good catch

**Review by adrianwyatt — APPROVED** [review]

File: samples/packaged/CopilotChat/WebApp/README.md (hunk 2 above)
Comment (adrianwyatt) [reply samples/packaged/CopilotChat/WebApp/README.md]: pushing to a future PR

**Pull request merged**